src/main/java/com/github/manolo8/darkbot/core/objects/IconGui.java

```diff
@@ -1,15 +1,26 @@
 package com.github.manolo8.darkbot.core.objects;
 
+import com.github.manolo8.darkbot.Main;
 import eu.darkbot.api.API;
 
 public class IconGui extends Gui implements API.Singleton {
+    @Override
+    public void update() {
+        if (address == 0) return;
+        super.update();
+        width = (int) Main.API.readMemoryDouble(address + 0x1F8);
+        height = (int) Main.API.readMemoryDouble(address + 0x200);
+        visible = Main.API.readMemoryLong(address + 0x180) != 0; // isVisible, sprite object is set
+    }
```

Comment:
the same update method exists in two guis, what's behind this decision?

Do all guis share this? is this a special type of gui only that does this?

Maybe the play here is a "PopupGui" class that extends gui and overrides update method like this, then icon gui & icon ok gui can extend PopupGui.